lib/server.js

```diff
@@ -1264,41 +1267,149 @@ WalletService.prototype._checkTxAndEstimateFee = function(txp) {
 
 WalletService.prototype._selectTxInputs = function(txp, utxosToExclude, cb) {
   var self = this;
+
   //todo: check inputs are ours and has enough value
   if (txp.inputs && txp.inputs.length > 0) {
     return cb(self._checkTxAndEstimateFee(txp));
   }
 
-  function sortUtxos(utxos) {
-    var list = _.map(utxos, function(utxo) {
-      var order;
-      if (utxo.confirmations == 0) {
-        order = 0;
-      } else if (utxo.confirmations < 6) {
-        order = -1;
-      } else {
-        order = -2;
-      }
-      return {
-        order: order,
-        utxo: utxo
-      };
-    });
-    return _.pluck(_.sortBy(list, 'order'), 'utxo');
-  };
-
-  self._getUtxosForCurrentWallet(null, function(err, utxos) {
-    if (err) return cb(err);
+  var txpAmount = txp.getTotalAmount();
+  var baseTxpSize = txp.getEstimatedSize();
+  var baseTxpFee = baseTxpSize * txp.feePerKb / 1000.;
+  var sizePerInput = txp.getEstimatedSizeForSingleInput();
+  var feePerInput = sizePerInput * txp.feePerKb / 1000.;
 
+  function sanitizeUtxos(utxos) {
     var excludeIndex = _.reduce(utxosToExclude, function(res, val) {
       res[val] = val;
       return res;
     }, {});
 
-    utxos = _.reject(utxos, function(utxo) {
-      return excludeIndex[utxo.txid + ":" + utxo.vout];
+    return _.filter(utxos, function(utxo) {
+      if (utxo.locked) return false;
+      if (utxo.satoshis <= feePerInput) return false;
```

Comment:
yes! we desperately needed this! as a side note, the choice of excluding utxos equal to the increase in fee they create is that they could become useful in a lower fee/kb setting.